src/content/en/fundamentals/engage-and-retain/push-notifications/web-push-protocol.md

```diff
@@ -0,0 +1,613 @@
+    
+With these headers set, we need to send the encrypted payload as the body of
+our request. Notice that the Content-Type is set to
+'application/octet-stream'. This is because the encrypted payload must be sent as a stream of
+bytes.
+
+In NodeJS we would do this like so:
+
+    const pushRequest = https.request(httpsOptions, function(pushResponse) {
+    pushRequest.write(encryptedPayload);
+    pushRequest.end();
+    
+## More Headers?
+
+We've covered the headers used for JWT / Application Server Keys (i.e. how to identify the
+application with the push service) and we've covered the headers used to send an encrypted
+payload.
+
+There are some additional headers that can be used by Push Services to alter the behavior of
+the send messages. Some of these headers are required, while others are optional.
+
+### TTL Header
+
+This is a **required header**.
+
+TTL (or time to live) should be an integer for the number of seconds you
+want your push message to live on the push service before it's delivered. If
+the TTL is expired, the message will be removed from the Push Service queue
+and it won't be delivered.
+
+    TTL: <Time to live in seconds>
+
+If you set a TTL of zero, the push service will attempt to deliver the message
+immediately **but** if the device can't be reached, your message will be
+immediately dropped from the push service queue.
+
+Technically a push service can reduce the TTL of a push message if it wants. You can tell if
+this has happened by examining the TTL header in the response from a push service.
+
+##### Topic
+
+This is an **optional header**.
+
+Topics are strings that can be used to replace any pending notifications with
+new notifications if they have matching topic names.
+
+This is useful in scenarios where the user's device might have been offline during which you
+might have sent three messages and you may want the user to receive only the latest message
+when their device is eventually turned back on.
+
+##### Urgency
+
+This is an **optional header**.
+
+This can be used to indicate to the push service how important a message is to the user. This
+could be used by the push service to help conserve the battery life of a users device by only
+waking up for important messages when battery is low.
+
+The header value should a string value of "very-low" | "low" | "normal" | "high". The default
+value is "normal".
+
+    Urgency: <very-low | low | normal | high>
+
+### Everything Together
+
+If you have further questions about how this all works you can always see how libraries trigger
+push messages on [the web-push-libs org](https://github.com/web-push-libs).
+
+Once you have an encrypted payload, and the headers above, you just need to make a POST request
+to the `endpoint` in a `PushSubscription`.
+
+So what do we do with the response to this POST request?
+
+### Response from Push Service
+
+Once you've made a request to a Push Service, you need to check the status code
```

Comment:
'Push Service' is not a proper noun. Make it lower case. Please check throughout.